src/main/java/com/eerussianguy/blazemap/feature/maps/WorldMapGui.java

```diff
@@ -285,23 +285,42 @@ public void render(GuiGraphics graphics, int i0, int i1, float f0) {
         }
     }
 
-    private void renderExportProgress(GuiGraphics graphics, float scale) {
+    private void renderAtlasExportProgress(GuiGraphics graphics, float scale) {
         AtlasExporter.Task task = AtlasExporter.getTask();
         if(task == null) return;
         Font font = Minecraft.getInstance().font;
         var stack = graphics.pose();
         stack.pushPose();
-        stack.translate(width - 205, 5, 0);
-        RenderHelper.fillRect(stack.last().pose(), 200, 30, Colors.WIDGET_BACKGROUND);
+
+        stack.translate(width - 205, 5, 0); // Go to corner
+        RenderHelper.fillRect(stack.last().pose(), 200, 30, Colors.WIDGET_BACKGROUND); // draw background
+
+        // Process flashing "animation"
+        int textColor = Colors.WHITE;
+        long flashUntil = ((long)task.getFlashUntil()) * 1000L;
+        long now = System.currentTimeMillis();
+        if(task.isErrored() || (flashUntil >= now && now % 333 < 166)) {
+            textColor = 0xFFFF0000;
+        }
+
+        // Render progress text
         int total = task.getTilesTotal();
         int current = task.getTilesCurrent();
+        graphics.drawString(font, "Exporting", 5, 5, textColor);
+        String operation = switch(task.getStage()){
+            case QUEUED -> "queued";
+            case CALCULATING -> "calculating";
+            case STITCHING -> String.format("stitching %d / %d tiles", current, total);
+            case SAVING -> "saving";
```

Comment:
Super duper low priority, but I just realised these should probably be translation strings. We can fix that up later though if you don't feel like extracting them to the JSON file

Reply:
Yes, I'll leave those for later.